### PR TITLE
1.7.0 doc updates

### LINKED ISSUE #501: [Enhancement] Update Documentation for CLI test Cmd
### Description

1. Add Documentation For Mutate Policy.
2. Add Documentation for NamespaceSelector in values. yaml
3. Update kyverno-test.yaml for validate policy. Need to add Namespace field in result section which validate against the 
   specific namespace.
  https://github.com/kyverno/kyverno/blob/d97258654f4bcb1f3e85e7531df22de675a82a6e/test/cli/test/any-all-wildcard/kyverno-test.yaml#L11
4. Update the result table section to following format:
```
│───│───────────────────────────────│──────────│──────────────────────│────────│
│ # │ POLICY                        │ RULE     │ RESOURCE             │ RESULT │
│───│───────────────────────────────│──────────│──────────────────────│────────│
│ 1 │ disallow-protected-namespaces │ disallow │ namespace1/Pod/test1 │ Pass   │
│ 2 │ disallow-protected-namespaces │ disallow │ namespace2/Pod/test2 │ Pass   │
│ 3 │ disallow-protected-namespaces │ disallow │ namespace3/Pod/test3 │ Pass   │
│───│───────────────────────────────│──────────│──────────────────────│────────│
```

### Slack discussion

_No response_

### LINKED ISSUE #502: [Bug] Generate policies should specify apiVersion in its rule definition
### Page link

_No response_

### Description

Generate policies will be covered by the test command soon. See PR kyverno/kyverno#3456. The CLI requires that user mentions apiVersion in generate policy's rule definition otherwise the test command would print out wrong result. Additionally, it is best practice to specify apiVersion along with kind. We should add this extra field in the website wherever we find an example of generate policy.

### Expected behavior

apiVersion specified in rule definition of generate policies.

### Slack discussion

_No response_

### LINKED ISSUE #519: [Enhancement] Document in line variable definition
### Description

See https://github.com/kyverno/kyverno/pull/3658 

### Slack discussion

_No response_

### LINKED ISSUE #527: [Enhancement] Remove YAML multiline docs
### Description

Once https://github.com/kyverno/kyverno/pull/3721 is merged, remove portions in the doc.

### Slack discussion

_No response_

### LINKED ISSUE #528: [Enhancement] kyverno jp is able to take in yaml docs as well apart from JSON docs
### Description

See https://github.com/kyverno/kyverno/pull/3768

### Slack discussion

_No response_

### LINKED ISSUE #532: Doc updates to cert rotation behavior
### Description

Need to document how certs are rotated upon expiry both for BYO and internal certs vis-a-vis the work of @eddycharly 

### Slack discussion

_No response_

### LINKED ISSUE #533: Document subresource format
### Description

In 1.7 subresources will be specifyable using either `/` or `.` notation between resource and subresource. Need to document this especially if it's just `.`.

### Slack discussion

_No response_

### LINKED ISSUE #535: [Enhancement] Document --allowInsecureRegistry parameter.
### Description

Once this [PR](https://github.com/kyverno/kyverno/pull/3983) would be merged we need to document `--allowInsecureRegistry` parameter.

### Slack discussion

_No response_

### LINKED ISSUE #544: [Enhancement] Document --autogenInternals flag
### Description

This new flag [appears](https://github.com/kyverno/kyverno/blob/704dc46ec30aeb548cd2ad6aa4c02c39829a4823/charts/kyverno/values.yaml#L144) in chart 2.4.0 but is not [documented](https://kyverno.io/docs/installation/#container-flags).

### Slack discussion

_No response_

### PR DESCRIPTION
**Pull request by chipzoller**
## Related issue #

Closes #544
Closes #535 
Closes #533 
Closes #532 
Closes #528 
Closes #527 
Closes #519 
Closes #502 
Closes #501 
## Proposed Changes

* Adds most of the issues addressed in 1.7.0. Will be updated later as time allows.
* Manually copies-and-pastes the content from https://github.com/kyverno/website/commit/733be7cd6bce6a62eeb01a6197b0460d9e872138 to present in `main`

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/website/blob/main/CONTRIBUTING.md).
- [x] I have inspected the website preview for accuracy.
- [x] I have signed off my issue.

### CODE REVIEW

**Comment by samj1912** [comment]
We should also cherry pick https://github.com/kyverno/website/commit/733be7cd6bce6a62eeb01a6197b0460d9e872138 back to main

File: content/en/docs/Writing policies/variables.md [hunk 1]
```diff
@@ -324,6 +324,36 @@ Note that certain characters must be escaped for JMESPath processing (ex. `-` in
...
+Variables may be defined in a `context` for consumption by Kyverno rules. This can be as simple as a static value or an object. The below sets a context variable with a value of `foo`.
+
+```yaml
+    context:
+    # A unique name for the for the variable 
+    # if the user redeclares a variable with the same name it should be re-assigned
+    - name: foodata
+      variable:
+        # value defines the value that the variable must have, it may contain jmespath variables or any yaml object that can be represented as a json object
+        value: "foo"
```
[the remark this answers is not in the document]
Comment (samj1912) [reply content/en/docs/Writing policies/variables.md:338]: See the example policy at https://github.com/kyverno/kyverno/blob/fe3c12628c9356eadf25c48717db9b9bf9e9eac8/test/cli/test/context-entries/policies.yaml#L40 and https://github.com/kyverno/kyverno/blob/fe3c12628c9356eadf25c48717db9b9bf9e9eac8/test/cli/test/context-entries/policies.yaml#L72

**Comment by samj1912** [comment]
> > We should also cherry pick [733be7c](https://github.com/kyverno/website/commit/733be7cd6bce6a62eeb01a6197b0460d9e872138) back to main
> 
> Thanks for that! I have manually copied-and-pasted (with minor updates) that text to this PR.

Should we also add an example with dockerhub as a registry? I know that was useful in a recent slack conversation.

For eg, we can use busybox as the image and show the variable values (index.docker.io/library/busybox for eg)

File: content/en/docs/Writing policies/variables.md [hunk 2]
```diff
@@ -324,6 +324,36 @@ Note that certain characters must be escaped for JMESPath processing (ex. `-` in
...
+
+Variables may be defined in a `context` for consumption by Kyverno rules. This can be as simple as a static value or an object. The below sets a context variable with a value of `foo`.
+
+```yaml
+    context:
+    # A unique name for the for the variable 
+    # if the user redeclares a variable with the same name it should be re-assigned
+    - name: foodata
+      variable:
+        # value defines the value that the variable must have, it may contain jmespath variables or any yaml object that can be represented as a json object
```
Comment (samj1912) [inline content/en/docs/Writing policies/variables.md:337]: All 3 fields (value, default and jmespath) are optional but either `value` or `jmespath` must be defined. `default` can only be used if `jmespath` is defined. All these combinations would automatically be validated during policy creation.

File: content/en/docs/Writing policies/variables.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (samj1912) [reply content/en/docs/Writing policies/variables.md:338]: The above file has examples for all the valid ways to define a new variable.

File: content/en/docs/Writing policies/variables.md (hunk 2 above)
Comment (chipzoller) [reply content/en/docs/Writing policies/variables.md]: Expanded comment to capture this. Resolve if this works.

**Comment by chipzoller** [comment]
> > > We should also cherry pick [733be7c](https://github.com/kyverno/website/commit/733be7cd6bce6a62eeb01a6197b0460d9e872138) back to main
> > 
> > 
> > Thanks for that! I have manually copied-and-pasted (with minor updates) that text to this PR.
> 
> Should we also add an example with dockerhub as a registry? I know that was useful in a recent slack conversation.
> 
> For eg, we can use busybox as the image and show the variable values (index.docker.io/library/busybox for eg)

Can I take this as a future action? There are other outstanding things I need to add but I'm out of time right now.

**Comment by chipzoller** [comment]
@samj1912 are you good with all comments? If not, can you resolve the ones that you are?

File: content/en/docs/Writing policies/variables.md [hunk 3]
```diff
@@ -339,6 +339,15 @@ Variables may be defined in a `context` for consumption by Kyverno rules. This c
         value: "foo"
 ```
 
+This snippet sets a context variable to the value of `request.object.metadata.name`.
+
+```yaml  - name: defined-jmespath
+context:
+- name: objName
+  variable:
+    jmesPath: request.object.metadata.name
```
Comment (samj1912) [inline content/en/docs/Writing policies/variables.md:348]: We should mention that if `value` is not defined, `jmespath` acts on the entire context (you can reference any variable here, request.*, previously defined variables etc.)

File: content/en/docs/Writing policies/variables.md [hunk 4]
```diff
@@ -326,7 +326,7 @@ You can also fetch image properties of all containers for further processing. Fo
 
 ## Inline Variables
 
-Variables may be defined in a `context` for consumption by Kyverno rules. This can be as simple as a static value or an object. The below sets a context variable with a value of `foo`.
+Variables may be defined in a `context` for consumption by Kyverno rules. This can be as simple as a static value, another variable, or a nested object. The below sets a context variable with a value of `foo`.
```
Comment (samj1912) [inline content/en/docs/Writing policies/variables.md:329]: Another important thing to note is that variables can be redefined by using the same variable name. The last value that is set is used.

You can also reference the same variable when redefining its value.

File: content/en/docs/Writing policies/variables.md (hunk 3 above)
Comment (chipzoller) [reply content/en/docs/Writing policies/variables.md:348]: Captured in https://github.com/kyverno/website/pull/548/commits/41ddd7d3a431c2aff42e3c4bd1be198ee7f3a58b

File: content/en/docs/Writing policies/variables.md (hunk 4 above)
Comment (samj1912) [reply content/en/docs/Writing policies/variables.md:329]: Example https://github.com/kyverno/kyverno/blob/fe3c12628c9356eadf25c48717db9b9bf9e9eac8/test/cli/test/context-entries/policies.yaml#L125

File: content/en/docs/Writing policies/variables.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (samj1912) [reply content/en/docs/Writing policies/variables.md:339]: It is also possible to define the jmespath itself as a variable https://github.com/kyverno/kyverno/blob/fe3c12628c9356eadf25c48717db9b9bf9e9eac8/test/cli/test/context-entries/policies.yaml#L114

File: content/en/docs/Writing policies/variables.md (hunk 4 above)
Comment (chipzoller) [reply content/en/docs/Writing policies/variables.md]: Captured in https://github.com/kyverno/website/pull/548/commits/3f68311543a3e77b2d730b7570d5b00b8b100429

File: content/en/docs/Writing policies/variables.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chipzoller) [reply content/en/docs/Writing policies/variables.md:339]: Captured in https://github.com/kyverno/website/pull/548/commits/509dab73be407d52bf237416f294a058d00c05b9

**Pull request merged**